test/components/Modal.test.tsx

```diff
@@ -12,13 +12,24 @@ afterEach(() => {
   expect(consoleError).not.toHaveBeenCalled();
 });
 
+interface ContextType {
+  foo: string
+};
+
+// `.isRequired` makes the test fail if context is not provided
+const FooType = React.PropTypes.string.isRequired;
+class ComponentAccessingContext extends React.Component {
+  static contextTypes = { foo: FooType };
+
+  render() {
+    return <div>{this.context.foo}</div>;
+  }
+}
+
 describe('Modal', () => {
 
   it('should correctly pass context down if used via modalWithContext', () => {
-    // `.isRequired` makes the test fail if context is not provided
-    const FooType = React.PropTypes.string.isRequired;
-    const ComponentAccessingContext = (_, ctx) => <div>{ctx.foo}</div>;
```

Comment:
Replaced with explicit class definition above. I don't know how to write types for `ComponentAccessingContext.contextTypes` otherwise.